lib/palindrome_check.rb

```diff
@@ -1,5 +1,44 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  # check for nil
+  unless my_phrase.nil?
+    # remove all space from phrase
+    my_phrase = remove_spaces(my_phrase)
+    # call method to reverse the phrase
+     return check_pal(my_phrase)
+  end
+
+  return false
+    # raise NotImplementedError
 end
+
+def remove_spaces(phrase)
+  i = 0
+  n = phrase.length
+  no_space_phrase = ''
+  while i < n
+    unless phrase[i] == ' '
+      no_space_phrase += phrase[i]
+    end
+    i += 1
+  end
+
+  return no_space_phrase
+end
+
+def check_pal(phrase)
+  i = 0
+  last = phrase.length - 1
+  flag = true
+  while i < last
```

Comment:
Can you think of a way to account for skipping consecutive white spaces in this loop? (Instead of altering the input string parameter by removing spaces in it.)

Reply:
Without altering the input parameter, I suppose I could check for spaces in the comparison method. So if phrase[i] == ' ' then increment i before comparing or if phrase[last] == ' ' then decrement last before comparing. In this case I'm guessing this would reduce my space complexity and potentially my time complexity. So no additional string would have to be stored and while iterating across the array in for comparison of i (start) and last (end) characters the space ( ' ') occurrence could be accounted for as they came up instead of necessitating that the entire string is iterated across first. For example in my implementation a string with no spaces will get iterated across to check for spaces and a new string gets stored even, if its an exact copy of the original. Taking your modified approach would definitely eliminate additional memory use and redundant iteration of the entire string.

Reply:
Nice! Good thinking!